Sources/ClientRuntime/Networking/Http/HttpBody.swift

```diff
@@ -26,13 +27,37 @@ public extension HttpBody {
         }
     }
 
+    /// Depending on the underlying type of the body, this will return a streamable object that can be used to stream the body.
+    /// - Returns: A streamable object that can be used to stream the body.
+    func toStreamable() -> IStreamable? {
+        switch self {
+        case let .data(data):
+            return data.map(ByteBuffer.init(data:))
+        case let .stream(stream):
+            switch stream {
+            case .buffer(let buffer):
+                return buffer
+            case .reader(let stream):
+                return HttpContent(stream: stream)
+            }
+        case .none:
+            return nil
+        }
+    }
+
     /// Returns true if the http body is `.none` or if the underlying data is nil or is empty.
+    /// For streamable bodies, this will return false if the stream is not empty.
     var isEmpty: Bool {
         switch self {
         case let .data(data):
             return data?.isEmpty ?? true
         case let .stream(stream):
-            return stream.toBytes().getData().isEmpty
+            switch stream {
+            case .buffer(let buffer):
+                return buffer.length() == 0
+            case .reader(let reader):
+                return reader.availableForRead != 0
```

Comment:
Nit: It'd be nice to define an `isEmpty` convenience on Stream. It would make our API more consistent with Swift and would make this snippet self-documenting. 

The snippet here would simply turn into: 
```swift 
return stream.isEmpty
```